tiledb/test/src/dimension.rs

```diff
@@ -0,0 +1,101 @@
+use num_traits::{Bounded, Num};
+use proptest::prelude::*;
+use std::fmt::Debug;
+use tiledb::array::{Dimension, DimensionBuilder};
+use tiledb::context::Context;
+use tiledb::fn_typed;
+use tiledb::Result as TileDBResult;
+
+use crate::strategy::LifetimeBoundStrategy;
+
+fn arbitrary_range_and_extent<T>() -> impl Strategy<Value = ([T; 2], T)>
```

Comment:
There are still some narrow cases where this strategy results in a panic... hopefully I can figure it out :/

Reply:
I think I got it.  For ints at least.  I'm definitely worried about floats.